backend/src/org/commcare/util/CommCareSession.java

```diff
@@ -721,6 +721,21 @@ public boolean isViewCommand(String command) {
         // commcare core for release issues
         return (entries.size() == 1 &&
                 (prototype.getXFormNamespace() == null ||
-                        prototype.getXFormNamespace().equals("")));
+                        prototype.getXFormNamespace().equals("")) &&
+                // Commcare-167204 by dcluna on 06/10/2015: If we have more than 1 datum for this Entry, we want to show them all
+                prototype.getSessionDataReqs().size() < 2);
```

Comment:
This should actually compare the session data requirements against the current set of session data.